lib/dialects/oracle/query.js

```diff
@@ -0,0 +1,142 @@
+'use strict';
+
+var Utils = require('../../utils')
+  , AbstractQuery = require('../abstract/query')
+  , _ = require('lodash')
+  , autocommit = true;
+
+var Query = function(connection, sequelize, options) {
+  this.connection = connection;
+  this.instance = options.instance;
+  this.model = options.model;
+  this.sequelize = sequelize;
+  this.options = _.extend({
+    logging: console.log,
+    plain: false,
+    raw: false
+  }, options || {});
+
+  this.checkLoggingOption();
+};
+
+Utils.inherit(Query, AbstractQuery);
+
+Query.prototype.run = function(sql) {
+  var self = this;
+  this.sql = sql;
+
+  this.sequelize.log('Executing (' + (this.connection.uuid || 'default') + '): ' + this.sql, this.options);
+
+  var promise = new Utils.Promise(function(resolve, reject) {
+    // TRANSACTION SUPPORT
+    if (_.contains(self.sql, 'ROLLBACK;')) {
+      console.log('should roll back');
+      self.connection.rollback(function(err) {
```

Comment:
Can we use nodeify instead? `resolve(Promise.nodeify(self.connection.rollback)())`

If we can, we might want to save a nodeifiied copy of the function, instead of nodeifying each time